### PR TITLE
[Fix issue 244] fix bug in gdr_generate_mr_handle

### LINKED ISSUE #244: gdrdrv returns the same handle for two pinned buffers
Under the following conditions, gdrdrv returns the same handle for two different pinned buffers:
1. The size of the first buffer is smaller than the GPU page.
2. The start of the second buffer is within the same page of the first buffer.

The handle for the second buffer supersedes that of the first buffer. One scenario that this behavior becomes problematic is:
- The first buffer is page-aligned.
- The second buffer is not page-aligned.

This will run correctly.
```
gdr_pin_buffer(g, first_buffer_ptr, first_buffer_size, 0, 0, &first_buffer_handle);
gdr_map(g, first_buffer_handle, &first_buffer_map_ptr, first_buffer_size);
gdr_pin_buffer(g, second_buffer_ptr, second_buffer_size, 0, 0, &second_buffer_handle);
```

This will fail because the offset of the second buffer is non-zero. But the failure happens when trying to map the first buffer!
```
gdr_pin_buffer(g, first_buffer_ptr, first_buffer_size, 0, 0, &first_buffer_handle);
gdr_pin_buffer(g, second_buffer_ptr, second_buffer_size, 0, 0, &second_buffer_handle);
gdr_map(g, first_buffer_handle, &first_buffer_map_ptr, first_buffer_size);
```

Code in #242 can be used to reproduce this bug.

### PR DESCRIPTION
**Pull request by pakmarkthub**
Issue:
- fixes #244.

This PR:
- introduces basic_small_buffers_mapping unit test in sanity.
- fixes bug in gdr_generate_mr_handle.
- bonus: fixes gdrdrv/Makefile to detect changes in gdrdrv.c.

Presubmit testing:
- locally on gc11 (x86).

### CODE REVIEW

File: src/gdrdrv/Makefile [hunk 1]
```diff
@@ -45,7 +45,7 @@ REL_MINOR  := $(word 3,$(REL))
 
 all: gdrdrv.ko
 
-gdrdrv.ko:
+gdrdrv.ko: gdrdrv.c
```
Comment (drossetti) [inline src/gdrdrv/Makefile:48]: can we remove this one? it will be addressed in a separate PR

**Review by drossetti — CHANGES_REQUESTED** [review]
one small nit

File: src/gdrdrv/Makefile (hunk 1 above)
Comment (pakmarkthub) [reply src/gdrdrv/Makefile]: Ok. Removed.

**Review by drossetti — APPROVED** [review]
nice catch!
